### PR TITLE
AAP-89067: Fix stale image purge when EE image reference form changes

### PR DESCRIPTION
**Pull request by djulich**
## Summary

- The `_handle_image_cleanup` signal (introduced in 2.6) compared EE images by string only, so two references to the same underlying image (e.g. `ee-minimal-rhel9:latest` vs `ee-minimal-rhel9@sha256:...`) were treated as different. This caused `podman rmi` to strip all names from a still-needed image, breaking environments — especially in disconnected installs where the image can't be re-pulled.
- Adds layered guards: skip cleanup for managed images (settings and `managed=True` EE objects), detect tag-vs-digest reference form changes for the same repository, and check other EEs for cross-reference form conflicts on delete.
- Genuinely superseded images (different tags like `v1` → `v2`) are still cleaned up.

## Test plan

- [x] Unit tests for `_extract_image_repo` (12 parametrized cases including registry ports, digests, bare names, edge cases)
- [x] Unit tests for `_different_ref_forms` (5 cases: tag↔digest, both tags, both digests, bare↔digest)
- [x] Functional test: managed image in `GLOBAL_JOB_EXECUTION_ENVIRONMENTS` → not cleaned
- [x] Functional test: image matching `CONTROL_PLANE_EXECUTION_ENVIRONMENT` → not cleaned
- [x] Functional test: `managed=True` EE object → not cleaned
- [x] Functional test: tag → digest same repo → not cleaned (AC #1)
- [x] Functional test: digest → tag same repo → not cleaned (AC #1)
- [x] Functional test: different tags same repo → **cleaned** (AC #2)
- [x] Functional test: delete EE when another EE shares repo with different ref form → not cleaned
- [x] Functional test: delete EE with no shared repos → cleaned
- [x] All 4 pre-existing tests still pass
- [ ] Backport to `stable-2.6` and `stable-2.7`

## Classification

Bug, Docs Fix or other nominal change

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved execution environment image cleanup to prevent removal of images still referenced by global, control-plane, or managed execution environments.
  - Preserved images when tag and digest references identify the same repository, including use across different reference formats.
  - Continued cleanup for genuinely unused images while avoiding removal of images shared across execution environments.
  - Correctly distinguishes different tags from tag-versus-digest references when determining whether an image can be safely removed.

### CODE REVIEW

File: awx/main/signals.py [hunk 1]
```diff
@@ -593,9 +593,99 @@ def deny_orphaned_approvals(sender, instance, **kwargs):
...
+            logger.info(
+                "Skipping cleanup of %s - new image %s appears to be a different reference form for the same image",
+                removed_image,
+                new_image,
+            )
+            return
+
+    # Delete path: another EE references the same repo with a different reference
+    # form, so purging could strip names from a still-needed image.
+    if removed_repo and _other_ee_shares_repo_with_different_ref(removed_image, removed_repo, pk):
```
[the remark this answers is not in the document]
Comment (djulich) [reply awx/main/signals.py:704]: Good catch - fixed.

**Comment by djulich** [comment]
YOLO (all test suites): https://jenkins-csb-aap-main.dno.corp.redhat.com/job/AAPQA/job/AAPQA%20Provisioner/job/Baby%20Yolo/job/AAPQA-Containerized-Baby-Yolo/521/

**Comment by djulich** [comment]
YOLO (controller atf only): https://jenkins-csb-aap-main.dno.corp.redhat.com/job/AAPQA/job/AAPQA%20Provisioner/job/Baby%20Yolo/job/AAPQA-Containerized-Baby-Yolo/520/

**Review by Jaapis — APPROVED** [review]

**Comment by djulich** [comment]
## Jenkins CI Analysis

Ran [Baby Yolo build #521](https://jenkins-csb-aap-main.dno.corp.redhat.com/job/AAPQA/job/AAPQA%20Provisioner/job/Baby%20Yolo/job/AAPQA-Containerized-Baby-Yolo/521/) (ALL TESTS, containerized cont_a topology) against this branch. 3227 of 6639 tests failed — **all with the same error:**

```
awxkit.exceptions.Unauthorized: Unauthorized (401) - 'Authentication credentials were not provided.'
```

This is a test environment auth setup issue, not a code regression. The test harness couldn't establish a login session, so every API test failed before executing.

[Build #520](https://jenkins-csb-aap-main.dno.corp.redhat.com/job/AAPQA/job/AAPQA%20Provisioner/job/Baby%20Yolo/job/AAPQA-Containerized-Baby-Yolo/520/) (same branch, smaller test scope) had only 7 failures, none related to our changes:

| Test | Area | Notes |
|------|------|-------|
| `test_azure_dynamic_inv` | Azure inventory | Unrelated |
| `test_session_cookie_age_change_affects_active_sessions` | Gateway auth | Unrelated |
| `test_explanations_complex_playbook` | Lightspeed | Unrelated |
| `test_explanations_without_explanation_id` | Lightspeed | Unrelated |
| `test_notification_inheritance` | Notifications | Pre-existing (since #518) |
| `test_config_get` | Licensing | Pre-existing (since #518) |
| `test_jt_instance_group` | Instance groups | Pre-existing (257 builds) |

The EE-specific test (`tests.controller.execution_environments.test_quay_ee_pub`) **passed**. No failures are related to EE image cleanup.
